### PR TITLE
Adressing proxy promotion issues

### PR DESCRIPTION
**Pull request by Patryk-Stefanski**
# What 

Addressing issues raised in this [comment ](https://issues.redhat.com/browse/THREESCALE-7135)

# Verification 

1. Deploy 3scale-operator locally 

2.Create dummy smtp secret
```sh
kubectl apply -f - <<EOF
---
kind: Secret
apiVersion: v1
metadata: 
  name: s3-credentials
  namespace: 3scale-test
data: 
  AWS_ACCESS_KEY_ID: UkVQTEFDRV9NRQ==
  AWS_BUCKET: UkVQTEFDRV9NRQ==
  AWS_REGION: UkVQTEFDRV9NRQ==
  AWS_SECRET_ACCESS_KEY: UkVQTEFDRV9NRQ==
type: Opaque
EOF
```

3.Create apimanager resource
```sh
kubectl apply -f - <<EOF
---
apiVersion: apps.3scale.net/v1alpha1
kind: APIManager
metadata: 
  name: apimanager-sample
  namespace: 3scale-test
spec: 
  system: 
    fileStorage: 
      simpleStorageService: 
        configurationSecretRef: 
          name: s3-credentials
  wildcardDomain: <cluster_domain>
EOF
```
cluster domain example : `apps.pstefans.uq32.s1.devshift.org`

4. Create Backend resource
```sh
kubectl apply -f - <<EOF
---
apiVersion: capabilities.3scale.net/v1beta1
kind: Backend
metadata:
  name: backend1-cr
  namespace: 3scale-test
spec:
  mappingRules:
    - httpMethod: GET
      increment: 1
      last: true
      metricMethodRef: hits
      pattern: /
  name: backend1
  privateBaseURL: 'https://api.example.com'
  systemName: backend1
EOF
```

5. Create product resources with the backend referenced
```sh
kubectl apply -f - <<EOF
---
apiVersion: capabilities.3scale.net/v1beta1
kind: Product
metadata:
  name: product1-cr
  namespace: 3scale-test
spec:
  name: product1
  backendUsages:
    backend1:
      path: /
EOF
```

6. Create proxyconfigpromote resources  to promote product to staging.
```sh
kubectl apply -f - <<EOF
---
apiVersion: capabilities.3scale.net/v1beta1
kind: ProxyConfigPromote
metadata:
  name: product1-v1-staging
  namespace: 3scale-test
spec:
  productCRName: product1-cr
 production: true
EOF
```

7. In 3scale UI check that product1 configuration was promoted to staging and that the CR now correctly reflects the staging and production versions.

### CODE REVIEW

**Comment by eguzki** [comment]
Can you fix indenting issues like https://github.com/3scale/3scale-operator/pull/773/files#diff-b7a85d834496ef811bc135bcf19a9cb43d407ed592225b17396bc8bcaaeff8cbR17? Not a big deal, but things like this are very weird

```
| **Field**                  | **json field**       | **Type** | **Info**                                                                                                                                                   | **Required** |
|----------------------------|----------------------|----------|------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------|
```

File: controllers/capabilities/proxyconfigpromote_controller.go [hunk 1]
```diff
@@ -189,6 +189,8 @@ func (r *ProxyConfigPromoteReconciler) proxyConfigPromoteReconciler(proxyConfigP
 			if err != nil {
 				statusReconciler := NewProxyConfigPromoteStatusReconciler(r.BaseReconciler, proxyConfigPromote, "Failed", productIDStr, latestProductionVersion, latestStagingVersion, err)
 				return statusReconciler, err
+			} else {
```
Comment (eguzki) [inline controllers/capabilities/proxyconfigpromote_controller.go:192]: this "spaghetti code" is calling for a refactor :)

Comment (eguzki) [reply controllers/capabilities/proxyconfigpromote_controller.go:192]: not to be done in this PR, though

**Review by eguzki — CHANGES_REQUESTED** [review]

**Comment by eguzki** [comment]
verification steps have passed

File: doc/proxyConfigPromote-reference.md [hunk 2]
```diff
@@ -24,6 +27,30 @@ Generated using [github-markdown-toc](https://github.com/ekalinin/github-markdow
 | Production                 | `production`         | bool     | If true promotes to production, if false promotes to staging                                                                                               | No           |
 | DeleteCR                   | `deleteCR`           | bool     | If true deletes the resource after a succesfull promotion                                                                                                  | No           |                                                                            | No |
 
+#### Provider Account Reference
```
[the remark this answers is not in the document]
Comment (Patryk-Stefanski) [reply doc/proxyConfigPromote-reference.md:30]: good spot, added it in. Thanks.

**Review by eguzki — APPROVED** [review]

**Pull request merged**